File: api-java/samples/server/trinsic/Hosted.java
package id.trinsic;

import id.trinsic.api.SessionsApi;
import id.trinsic.api.models.CreateHostedProviderSessionRequest;
import id.trinsic.api.models.CreateHostedProviderSessionResponse;
import io.javalin.Javalin;
import java.util.UUID;
public class Hosted {
    public static void HostedRoutes(Javalin app, SessionsApi session, UUID verificationProfileId){

        app.post("/create-hosted-session/{provider}", ctx -> {
            // Get path parameter
            String provider = ctx.pathParam("provider");

            // Get query parameter
            String redirectUrl = ctx.queryParam("redirectUrl");

            // Build request object
            CreateHostedProviderSessionRequest req = new CreateHostedProviderSessionRequest();
            req.setRedirectUrl(redirectUrl);
            req.setProvider(provider);
            req.setVerificationProfileId(verificationProfileId);

            // Call the service
            CreateHostedProviderSessionResponse result = session.createHostedProviderSession(req);

            // Redirect to result URL
            ctx.json(result);
        });
    }
}

File: api-php/samples/server/src/hosted.php
<?php

use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;
use Trinsic\Api\Model\CreateHostedProviderSessionRequest as CreateHostedProviderSessionRequest;
use Trinsic\Api\Model\CreateHostedProviderSessionResponse as CreateHostedProviderSessionResponse;

return function ($app, $sessions) {
    

    $app->post("/create-hosted-session/{provider}", function (Request $request, Response $response, $args) use ($sessions) {
        $provider = $args['provider'];
        $redirectUrl = $request->getQueryParams()['redirectUrl'];
    
        $req = new CreateHostedProviderSessionRequest();
        $verificationProfileId = getenv('TRINSIC_VERIFICATION_PROFILE_ID');
        $req->setVerificationProfileId($verificationProfileId);
        $req->setRedirectUrl($redirectUrl);
        $req->setProvider($provider);
    
        $result = $sessions->createHostedProviderSession($req);
    
        $response->getBody()->write(json_encode($result));
        return $response->withHeader('Content-Type', 'application/json');
    });

};